### PR TITLE
Wf 868 implement tasks whenever a release workflow is triggered

### PR DESCRIPTION
**Pull request by manisha-backbase**

### CODE REVIEW

File: libs/transactions/src/lib/components/transaction-item/transaction-item.component.ts [hunk 1]
```diff
@@ -41,6 +41,7 @@ export class TransactionItemComponent implements OnChanges {
   }
 
   ngOnChanges(changes: SimpleChanges): void {
+    console.log("testing123");
```
Comment (Farzannajipour) [inline libs/transactions/src/lib/components/transaction-item/transaction-item.component.ts:44]: Please remove it.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "golden-sample-app",
-  "version": "0.0.0",
+  "version": "1.0.1",
```
Comment (Farzannajipour) [inline package.json:3]: Why the version is 1.0.1?

File: libs/transactions/src/lib/components/transaction-item/transaction-item.component.ts (hunk 1 above)
Comment (manisha-backbase) [reply libs/transactions/src/lib/components/transaction-item/transaction-item.component.ts]: Removed

File: package.json (hunk 2 above)
Comment (manisha-backbase) [reply package.json]: Removed

File: .github/actions/bump-version/action.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    - name: Push changes
+      shell: bash
+      run: git push origin HEAD
+    - name: Push tags
+      shell: bash
+      run: git push --tags
+    - name: Tag last-release
+      shell: bash
+      run: |
+          git tag -f last-release
```
Comment (kirillsud) [inline .github/actions/bump-version/action.yml:41]: I think the `last-release` tag should be different for a specific library. Can we use a library release tag instead, e.g. `lib1@vX.Y.Z`?

**Comment by santam85** [comment]
the workflow folder it's starting to become messy. Ideally what I'd like to see is just 3 workflows:

1. running at every pr created or updated, running the whole ci process againts the merge result
2. the release-drafter, running every time anything is pushed to main
3. a release workflow, running only when publishing a release. This should be the only place where we bump versions

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+          fetch-depth: 0
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: 'npm'
+          registry-url: 'https://repo.backbase.com/artifactory/api/npm/npm-backbase/'
+          scope: '@backbase'   
+      - name: Release
+        uses: ./.github/actions/bump-version
```
Comment (kirillsud) [inline .github/workflows/ci.yml:24]: Is it only about bumping versions or also about publishing packages?

**Review by kirillsud — DISMISSED** [review]

File: .github/workflows/ci.yml (hunk 4 above)
Comment (manisha-backbase) [reply .github/workflows/ci.yml:24]: Later on we can add a step for publishing packages as well

Comment (manisha-backbase) [reply .github/workflows/ci.yml:24]: The semver package has a post-target hook to publish the version on NPM, GitHub or GitLab.
https://github.com/jscutlery/semver#triggering-executors-post-release

File: .github/actions/bump-version/action.yml (hunk 3 above)
Comment (manisha-backbase) [reply .github/actions/bump-version/action.yml:41]: Here this last-release tag is used for computing the affected packages ` npx nx affected --base=last-release --parallel=1 --target=version`
And this last-release tag is always pushed whenever the CI workflow will be triggered for publishing any package.

**Comment by manisha-backbase** [comment]
> the workflow folder it's starting to become messy. Ideally what I'd like to see is just 3 workflows:
> 
> 1. running at every pr created or updated, running the whole ci process againts the merge result
> 2. the release-drafter, running every time anything is pushed to main
> 3. a release workflow, running only when publishing a release. This should be the only place where we bump versions

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+      #     scope: '@backbase'   
+      # - name: Release
+      #   uses: ./.github/actions/bump-version
+      #   with:
+      #     github_token: ${{ secrets.GITHUB_TOKEN }}
+      #     npm_token: ${{ secrets.NPM_TOKEN }}
+      - name: 'Extract project from tag'
+        id: set-project-from-tag
+        run: |
+          module="$(echo "$GITHUB_REF" | sed 's~^refs/tags/\(.*\)-v[0-9.]\+$~\1~')"
```
Comment (kirillsud) [inline .github/workflows/release.yml:32]: I believe you can use `${{ github.event.release.tag_name }}` to fetch the tag name

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+      # - name: Release
+      #   uses: ./.github/actions/bump-version
+      #   with:
+      #     github_token: ${{ secrets.GITHUB_TOKEN }}
+      #     npm_token: ${{ secrets.NPM_TOKEN }}
+      - name: 'Extract project from tag'
+        id: set-project-from-tag
+        run: |
+          module=$(echo "${{ github.event.release.tag_name }}" | awk -F"-v" '{print $1}')
+          version=$(echo "${{ github.event.release.tag_name }}" | awk -F"-v" '{print $2}')
```
Comment (kirillsud) [inline .github/workflows/release.yml:35]:
```suggestion
          version=$(echo "${{ github.event.release.tag_name }}" | awk -F"@v" '{print $2}')
```

Comment (kirillsud) [inline .github/workflows/release.yml:35]:
```suggestion
          module=$(echo "${{ github.event.release.tag_name }}" | awk -F"@v" '{print $1}')
          version=$(echo "${{ github.event.release.tag_name }}" | awk -F"@v" '{print $2}')
```

File: .github/workflows/release.yml [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+          echo "::set-output name=module::$module"
+          echo "::set-output name=version::$version"
+
+      
+      - name: 'Bump'
+        run: cd libs/${{ steps.set-project-from-tag.outputs.module }};
+             npm version ${{ steps.set-project-from-tag.outputs.version }}
+      
+      - name: Commit files
+        run: git commit -m "[version bump] Version bumped" -a
```
Comment (kirillsud) [inline .github/workflows/release.yml:46]: Can we add to a commit message a specific library and version we bumped?

File: .github/workflows/release.yml (hunk 6 above)
Comment (manisha-backbase) [reply .github/workflows/release.yml]: As the master config file had -v, I had kept it for testing purpose. Changed it to @v for the final PR

File: .github/workflows/release.yml (hunk 7 above)
Comment (manisha-backbase) [reply .github/workflows/release.yml]: Added

**Review by santam85 — APPROVED** [review]
Looks good to me!

**Review by Farzannajipour — APPROVED** [review]

**Pull request merged**